{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "69bbf72b",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pybedtools import BedTool\n",
    "import pandas as pd\n",
    "import glob\n",
    "import os "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "bc847348",
   "metadata": {},
   "outputs": [],
   "source": [
    "# PARAMETERS\n",
    "input_dir = \"/anvil/projects/x-mcb130189/Wubin/BG/Browser/Tracks\"\n",
    "gwas_studies = \"/anvil/projects/x-mcb130189/rwang22/references/GWAS/signif_neuro_gwas_studies.csv\"\n",
    "genes_bed = \"/anvil/projects/x-mcb130189/rwang22/references/hg38/gencode.v48.gene.bed\"\n",
    "chrom_sizes = \"/anvil/projects/x-mcb130189/rwang22/references/hg38/hg38.chrom.sizes\"\n",
    "outfile = \"abc_signif_snps_overlap.tsv\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "4283718e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# snp : rsid - rs1950834\n",
    "# chr14:41605321\n",
    "# make the snps dataframe \n",
    "snp_df = pd.DataFrame({\n",
    "    \"chrom\": [\"chr14\"], \n",
    "    \"start\": [41605320], \n",
    "    \"end\": [41605321]\n",
    "})\n",
    "\n",
    "snp = BedTool.from_dataframe(snp_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "01647cbb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNPS</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>P</th>\n",
       "      <th>trait</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17.0</td>\n",
       "      <td>rs11870683</td>\n",
       "      <td>39973588.0</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>2.786000e-08</td>\n",
       "      <td>Mullins.NatGenet.2021.Bipolar_Disorder.tsv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>17.0</td>\n",
       "      <td>rs61554907</td>\n",
       "      <td>40064179.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>1.636000e-08</td>\n",
       "      <td>Mullins.NatGenet.2021.Bipolar_Disorder.tsv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>15.0</td>\n",
       "      <td>rs1894401</td>\n",
       "      <td>90885812.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2.799000e-08</td>\n",
       "      <td>Mullins.NatGenet.2021.Bipolar_Disorder.tsv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>15.0</td>\n",
       "      <td>rs17514846</td>\n",
       "      <td>90873320.0</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>4.765000e-08</td>\n",
       "      <td>Mullins.NatGenet.2021.Bipolar_Disorder.tsv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>15.0</td>\n",
       "      <td>rs6224</td>\n",
       "      <td>90880313.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>2.877000e-08</td>\n",
       "      <td>Mullins.NatGenet.2021.Bipolar_Disorder.tsv</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHR        SNPS          BP A1 A2             P  \\\n",
       "0  17.0  rs11870683  39973588.0  T  A  2.786000e-08   \n",
       "1  17.0  rs61554907  40064179.0  G  T  1.636000e-08   \n",
       "2  15.0   rs1894401  90885812.0  G  A  2.799000e-08   \n",
       "3  15.0  rs17514846  90873320.0  C  A  4.765000e-08   \n",
       "4  15.0      rs6224  90880313.0  G  T  2.877000e-08   \n",
       "\n",
       "                                        trait  \n",
       "0  Mullins.NatGenet.2021.Bipolar_Disorder.tsv  \n",
       "1  Mullins.NatGenet.2021.Bipolar_Disorder.tsv  \n",
       "2  Mullins.NatGenet.2021.Bipolar_Disorder.tsv  \n",
       "3  Mullins.NatGenet.2021.Bipolar_Disorder.tsv  \n",
       "4  Mullins.NatGenet.2021.Bipolar_Disorder.tsv  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brain_gwas = pd.read_csv(gwas_studies, index_col = 0).rename(columns = {\"SNP\": \"SNPS\"})\n",
    "brain_gwas.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "35e3d8f3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Mullins.NatGenet.2021.Bipolar_Disorder.tsv',\n",
       "       'Bellenguez.NatGenet.2022.Alzheimers_Disease_Dementia.h.tsv',\n",
       "       'Grove.NatGenet.2019.Autism_Spectrum_Disorder.tsv',\n",
       "       'Wray.NatGenet.2018.MDD.tsv',\n",
       "       'Demontis.NatGenet.2019.Attention_Deficit_Hyperactivity_Disorder.tsv',\n",
       "       'Nalls.LancetNeurol.2019.Parkinsons_disease.h.tsv',\n",
       "       'PGC.Nature.2014.Schizophrenia.tsv'], dtype=object)"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brain_gwas[\"trait\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "f94fece1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR_ID</th>\n",
       "      <th>START</th>\n",
       "      <th>END</th>\n",
       "      <th>SNPS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr17</td>\n",
       "      <td>39973587</td>\n",
       "      <td>39973588</td>\n",
       "      <td>rs11870683</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr17</td>\n",
       "      <td>40064178</td>\n",
       "      <td>40064179</td>\n",
       "      <td>rs61554907</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr15</td>\n",
       "      <td>90885811</td>\n",
       "      <td>90885812</td>\n",
       "      <td>rs1894401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr15</td>\n",
       "      <td>90873319</td>\n",
       "      <td>90873320</td>\n",
       "      <td>rs17514846</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr15</td>\n",
       "      <td>90880312</td>\n",
       "      <td>90880313</td>\n",
       "      <td>rs6224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20730</th>\n",
       "      <td>chr12</td>\n",
       "      <td>2177796</td>\n",
       "      <td>2177797</td>\n",
       "      <td>rs6489351</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20737</th>\n",
       "      <td>chr12</td>\n",
       "      <td>2405103</td>\n",
       "      <td>2405104</td>\n",
       "      <td>rs714277</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20739</th>\n",
       "      <td>chr12</td>\n",
       "      <td>2402664</td>\n",
       "      <td>2402665</td>\n",
       "      <td>rs2239063</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20745</th>\n",
       "      <td>chr2</td>\n",
       "      <td>161960217</td>\n",
       "      <td>161960218</td>\n",
       "      <td>rs12472555</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20746</th>\n",
       "      <td>chr2</td>\n",
       "      <td>161989344</td>\n",
       "      <td>161989345</td>\n",
       "      <td>rs2909457</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17810 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHR_ID      START        END        SNPS\n",
       "0      chr17   39973587   39973588  rs11870683\n",
       "1      chr17   40064178   40064179  rs61554907\n",
       "2      chr15   90885811   90885812   rs1894401\n",
       "3      chr15   90873319   90873320  rs17514846\n",
       "4      chr15   90880312   90880313      rs6224\n",
       "...      ...        ...        ...         ...\n",
       "20730  chr12    2177796    2177797   rs6489351\n",
       "20737  chr12    2405103    2405104    rs714277\n",
       "20739  chr12    2402664    2402665   rs2239063\n",
       "20745   chr2  161960217  161960218  rs12472555\n",
       "20746   chr2  161989344  161989345   rs2909457\n",
       "\n",
       "[17810 rows x 4 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas_bed = brain_gwas[[\"CHR\", \"BP\", \"SNPS\"]].rename(columns = {\"BP\": \"END\", \"CHR\": \"CHR_ID\"})\n",
    "gwas_bed = gwas_bed[~gwas_bed[\"END\"].isna()] # variants with NaN position likely could not be mapped to hg38 -> discard \n",
    "gwas_bed[\"CHR_ID\"] = gwas_bed[\"CHR_ID\"].apply(lambda x: f\"chr{int(x)}\") \n",
    "gwas_bed[\"START\"] = gwas_bed[\"END\"].apply(lambda x: x - 1) \n",
    "gwas_bed[\"START\"] = gwas_bed[\"START\"].astype(int)\n",
    "gwas_bed[\"END\"] = gwas_bed[\"END\"].astype(int)\n",
    "gwas_bed = gwas_bed[[\"CHR_ID\", \"START\", \"END\", \"SNPS\"]].drop_duplicates()\n",
    "gwas_bed "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "4758a2e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "gwas = BedTool.from_dataframe(gwas_bed).sort()\n",
    "genes = BedTool(genes_bed).sort()\n",
    "closest = gwas.closest(genes, d=True).to_dataframe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "1a4c8df1",
   "metadata": {},
   "outputs": [],
   "source": [
    "closest = closest[[\"chrom\", \"start\", \"end\", \"name\", \"thickEnd\"]]\n",
    "closest = BedTool.from_dataframe(closest)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4a5f3b40",
   "metadata": {},
   "source": [
    "# group - bed files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "736f936a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR_ID</th>\n",
       "      <th>START</th>\n",
       "      <th>END</th>\n",
       "      <th>SNPS</th>\n",
       "      <th>Gene</th>\n",
       "      <th>trait</th>\n",
       "      <th>celltype</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2438058</td>\n",
       "      <td>2438059</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2438058</td>\n",
       "      <td>NaN</td>\n",
       "      <td>STR_SST-CHODL_GABA.bed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2443318</td>\n",
       "      <td>2443319</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2443318</td>\n",
       "      <td>NaN</td>\n",
       "      <td>STR_SST-CHODL_GABA.bed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2444404</td>\n",
       "      <td>2444405</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2444404</td>\n",
       "      <td>NaN</td>\n",
       "      <td>STR_SST-CHODL_GABA.bed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>8372075</td>\n",
       "      <td>8372076</td>\n",
       "      <td>chr1</td>\n",
       "      <td>8372075</td>\n",
       "      <td>NaN</td>\n",
       "      <td>STR_SST-CHODL_GABA.bed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>8409223</td>\n",
       "      <td>8409224</td>\n",
       "      <td>chr1</td>\n",
       "      <td>8409223</td>\n",
       "      <td>NaN</td>\n",
       "      <td>STR_SST-CHODL_GABA.bed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33714</th>\n",
       "      <td>chr22</td>\n",
       "      <td>41217298</td>\n",
       "      <td>41217299</td>\n",
       "      <td>chr22</td>\n",
       "      <td>41217298</td>\n",
       "      <td>NaN</td>\n",
       "      <td>F_M_Glut.bed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33715</th>\n",
       "      <td>chr22</td>\n",
       "      <td>41217298</td>\n",
       "      <td>41217299</td>\n",
       "      <td>chr22</td>\n",
       "      <td>41217298</td>\n",
       "      <td>NaN</td>\n",
       "      <td>F_M_Glut.bed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33716</th>\n",
       "      <td>chr22</td>\n",
       "      <td>42138116</td>\n",
       "      <td>42138117</td>\n",
       "      <td>chr22</td>\n",
       "      <td>42138116</td>\n",
       "      <td>NaN</td>\n",
       "      <td>F_M_Glut.bed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33717</th>\n",
       "      <td>chr22</td>\n",
       "      <td>42138116</td>\n",
       "      <td>42138117</td>\n",
       "      <td>chr22</td>\n",
       "      <td>42138116</td>\n",
       "      <td>NaN</td>\n",
       "      <td>F_M_Glut.bed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33718</th>\n",
       "      <td>chr22</td>\n",
       "      <td>42225996</td>\n",
       "      <td>42225997</td>\n",
       "      <td>chr22</td>\n",
       "      <td>42225996</td>\n",
       "      <td>NaN</td>\n",
       "      <td>F_M_Glut.bed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>33719 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHR_ID     START       END   SNPS      Gene trait  \\\n",
       "0       chr1   2438058   2438059   chr1   2438058   NaN   \n",
       "1       chr1   2443318   2443319   chr1   2443318   NaN   \n",
       "2       chr1   2444404   2444405   chr1   2444404   NaN   \n",
       "3       chr1   8372075   8372076   chr1   8372075   NaN   \n",
       "4       chr1   8409223   8409224   chr1   8409223   NaN   \n",
       "...      ...       ...       ...    ...       ...   ...   \n",
       "33714  chr22  41217298  41217299  chr22  41217298   NaN   \n",
       "33715  chr22  41217298  41217299  chr22  41217298   NaN   \n",
       "33716  chr22  42138116  42138117  chr22  42138116   NaN   \n",
       "33717  chr22  42138116  42138117  chr22  42138116   NaN   \n",
       "33718  chr22  42225996  42225997  chr22  42225996   NaN   \n",
       "\n",
       "                     celltype   A1   A2   P  \n",
       "0      STR_SST-CHODL_GABA.bed  NaN  NaN NaN  \n",
       "1      STR_SST-CHODL_GABA.bed  NaN  NaN NaN  \n",
       "2      STR_SST-CHODL_GABA.bed  NaN  NaN NaN  \n",
       "3      STR_SST-CHODL_GABA.bed  NaN  NaN NaN  \n",
       "4      STR_SST-CHODL_GABA.bed  NaN  NaN NaN  \n",
       "...                       ...  ...  ...  ..  \n",
       "33714            F_M_Glut.bed  NaN  NaN NaN  \n",
       "33715            F_M_Glut.bed  NaN  NaN NaN  \n",
       "33716            F_M_Glut.bed  NaN  NaN NaN  \n",
       "33717            F_M_Glut.bed  NaN  NaN NaN  \n",
       "33718            F_M_Glut.bed  NaN  NaN NaN  \n",
       "\n",
       "[33719 rows x 10 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dmr_overlap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "c0024fed",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Group\n",
      "  chrom    start      end  name     score strand thickStart  thickEnd  \\\n",
      "0  chr1  2438058  2438059   NaN  0.695848      .       chr1   2438058   \n",
      "1  chr1  2443318  2443319   NaN  0.867963      .       chr1   2443318   \n",
      "2  chr1  8408311  8408312   NaN  2.348902      .       chr1   8408311   \n",
      "3  chr1  8409223  8409224   NaN  2.978643      .       chr1   8409223   \n",
      "4  chr1  8409276  8409277   NaN  2.978643      .       chr1   8409276   \n",
      "\n",
      "   itemRgb blockCount blockSizes      celltype  \n",
      "0  2438059  rs4592207      PLCH2  Pericyte.bed  \n",
      "1  2443319  rs4648844      PLCH2  Pericyte.bed  \n",
      "2  8408312   rs301791       RERE  Pericyte.bed  \n",
      "3  8409224   rs301790       RERE  Pericyte.bed  \n",
      "4  8409277   rs301789       RERE  Pericyte.bed  \n",
      "Subclass\n",
      "  chrom    start      end  name     score strand thickStart  thickEnd  \\\n",
      "0  chr1  2438058  2438059   NaN  1.280416      .       chr1   2438058   \n",
      "1  chr1  2443318  2443319   NaN  1.079933      .       chr1   2443318   \n",
      "2  chr1  2444404  2444405   NaN  0.362516      .       chr1   2444404   \n",
      "3  chr1  8372075  8372076   NaN  1.896631      .       chr1   8372075   \n",
      "4  chr1  8409223  8409224   NaN  0.870363      .       chr1   8409223   \n",
      "\n",
      "   itemRgb blockCount blockSizes                celltype  \n",
      "0  2438059  rs4592207      PLCH2  STR_SST-CHODL_GABA.bed  \n",
      "1  2443319  rs4648844      PLCH2  STR_SST-CHODL_GABA.bed  \n",
      "2  2444405  rs6687012      PLCH2  STR_SST-CHODL_GABA.bed  \n",
      "3  8372076   rs894875       RERE  STR_SST-CHODL_GABA.bed  \n",
      "4  8409224   rs301790       RERE  STR_SST-CHODL_GABA.bed  \n"
     ]
    }
   ],
   "source": [
    "for p in [\"Group\", \"Subclass\"]:\n",
    "    print(p)\n",
    "    bed_files = {f: BedTool(f) for f in glob.glob(f'{input_dir}/{p}/ABC.links/*.bed')}\n",
    "    df_list = []\n",
    "    for bed in bed_files:\n",
    "            output = bed_files[bed].intersect(closest, wb = True)\n",
    "            if len(output) > 1:\n",
    "                tmp = output.to_dataframe()\n",
    "                filename = os.path.basename(bed)\n",
    "                tmp[\"celltype\"] = filename\n",
    "                df_list.append(tmp)\n",
    "\n",
    "    dmr_overlap = pd.concat(df_list, ignore_index = True)\n",
    "    print(dmr_overlap.head())\n",
    "    dmr_overlap = dmr_overlap[[\"chrom\", \"start\", \"end\", \"blockCount\", \"blockSizes\", \"celltype\"]].rename(columns = {\"chrom\": \"CHR_ID\", \n",
    "                                                                                            \"start\" : \"START\", \n",
    "                                                                                            \"end\": \"END\", \n",
    "                                                                                            \"blockCount\": \"SNPS\", \n",
    "                                                                                            \"blockSizes\" : \"Gene\"})\n",
    "    dmr_overlap = dmr_overlap.merge(brain_gwas, how = \"left\", on =\"SNPS\")[[\"CHR_ID\", \"START\", \"END\", \"SNPS\", \"Gene\", \n",
    "                                                                            \"trait\", \"celltype\", \"A1\", \"A2\", \"P\"]]\n",
    "\n",
    "    dmr_overlap.to_csv(f'genetic_variant/{p}.{outfile}.tsv', sep = \"\\t\", index = False)\n",
    " \n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.20"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
