{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "69bbf72b",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pybedtools import BedTool\n",
    "import pandas as pd\n",
    "import glob\n",
    "import os "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "bc847348",
   "metadata": {},
   "outputs": [],
   "source": [
    "# PARAMETERS\n",
    "input_dir = \"/anvil/projects/x-mcb130189/Wubin/BG/pseudobulk/DMR/Subclass\"\n",
    "gwas_studies = \"/anvil/scratch/x-rwang22/GWAS/signif_neuro_gwas_studies.csv\"\n",
    "genes_bed = \"/anvil/projects/x-mcb130189/rwang22/references/hg38/gencode.v48.gene.bed\"\n",
    "chrom_sizes = \"/anvil/projects/x-mcb130189/rwang22/references/hg38/hg38.chrom.sizes\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "4283718e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# snp : rsid - rs1950834\n",
    "# chr14:41605321\n",
    "# make the snps dataframe \n",
    "snp_df = pd.DataFrame({\n",
    "    \"chrom\": [\"chr14\"], \n",
    "    \"start\": [41605320], \n",
    "    \"end\": [41605321]\n",
    "})\n",
    "\n",
    "snp = BedTool.from_dataframe(snp_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "01647cbb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNPS</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>P</th>\n",
       "      <th>trait</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17.0</td>\n",
       "      <td>rs11870683</td>\n",
       "      <td>39973588.0</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>2.786000e-08</td>\n",
       "      <td>Mullins.NatGenet.2021.Bipolar_Disorder.tsv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>17.0</td>\n",
       "      <td>rs61554907</td>\n",
       "      <td>40064179.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>1.636000e-08</td>\n",
       "      <td>Mullins.NatGenet.2021.Bipolar_Disorder.tsv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>15.0</td>\n",
       "      <td>rs1894401</td>\n",
       "      <td>90885812.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2.799000e-08</td>\n",
       "      <td>Mullins.NatGenet.2021.Bipolar_Disorder.tsv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>15.0</td>\n",
       "      <td>rs17514846</td>\n",
       "      <td>90873320.0</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>4.765000e-08</td>\n",
       "      <td>Mullins.NatGenet.2021.Bipolar_Disorder.tsv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>15.0</td>\n",
       "      <td>rs6224</td>\n",
       "      <td>90880313.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>2.877000e-08</td>\n",
       "      <td>Mullins.NatGenet.2021.Bipolar_Disorder.tsv</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHR        SNPS          BP A1 A2             P  \\\n",
       "0  17.0  rs11870683  39973588.0  T  A  2.786000e-08   \n",
       "1  17.0  rs61554907  40064179.0  G  T  1.636000e-08   \n",
       "2  15.0   rs1894401  90885812.0  G  A  2.799000e-08   \n",
       "3  15.0  rs17514846  90873320.0  C  A  4.765000e-08   \n",
       "4  15.0      rs6224  90880313.0  G  T  2.877000e-08   \n",
       "\n",
       "                                        trait  \n",
       "0  Mullins.NatGenet.2021.Bipolar_Disorder.tsv  \n",
       "1  Mullins.NatGenet.2021.Bipolar_Disorder.tsv  \n",
       "2  Mullins.NatGenet.2021.Bipolar_Disorder.tsv  \n",
       "3  Mullins.NatGenet.2021.Bipolar_Disorder.tsv  \n",
       "4  Mullins.NatGenet.2021.Bipolar_Disorder.tsv  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brain_gwas = pd.read_csv(gwas_studies, index_col = 0).rename(columns = {\"SNP\": \"SNPS\"})\n",
    "brain_gwas.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "35e3d8f3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Mullins.NatGenet.2021.Bipolar_Disorder.tsv',\n",
       "       'Bellenguez.NatGenet.2022.Alzheimers_Disease_Dementia.h.tsv',\n",
       "       'Grove.NatGenet.2019.Autism_Spectrum_Disorder.tsv',\n",
       "       'Wray.NatGenet.2018.MDD.tsv',\n",
       "       'Demontis.NatGenet.2019.Attention_Deficit_Hyperactivity_Disorder.tsv',\n",
       "       'Nalls.LancetNeurol.2019.Parkinsons_disease.h.tsv',\n",
       "       'PGC.Nature.2014.Schizophrenia.tsv'], dtype=object)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brain_gwas[\"trait\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "f94fece1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR_ID</th>\n",
       "      <th>START</th>\n",
       "      <th>END</th>\n",
       "      <th>SNPS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr17</td>\n",
       "      <td>39973587</td>\n",
       "      <td>39973588</td>\n",
       "      <td>rs11870683</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr17</td>\n",
       "      <td>40064178</td>\n",
       "      <td>40064179</td>\n",
       "      <td>rs61554907</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr15</td>\n",
       "      <td>90885811</td>\n",
       "      <td>90885812</td>\n",
       "      <td>rs1894401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr15</td>\n",
       "      <td>90873319</td>\n",
       "      <td>90873320</td>\n",
       "      <td>rs17514846</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr15</td>\n",
       "      <td>90880312</td>\n",
       "      <td>90880313</td>\n",
       "      <td>rs6224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20730</th>\n",
       "      <td>chr12</td>\n",
       "      <td>2177796</td>\n",
       "      <td>2177797</td>\n",
       "      <td>rs6489351</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20737</th>\n",
       "      <td>chr12</td>\n",
       "      <td>2405103</td>\n",
       "      <td>2405104</td>\n",
       "      <td>rs714277</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20739</th>\n",
       "      <td>chr12</td>\n",
       "      <td>2402664</td>\n",
       "      <td>2402665</td>\n",
       "      <td>rs2239063</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20745</th>\n",
       "      <td>chr2</td>\n",
       "      <td>161960217</td>\n",
       "      <td>161960218</td>\n",
       "      <td>rs12472555</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20746</th>\n",
       "      <td>chr2</td>\n",
       "      <td>161989344</td>\n",
       "      <td>161989345</td>\n",
       "      <td>rs2909457</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17810 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHR_ID      START        END        SNPS\n",
       "0      chr17   39973587   39973588  rs11870683\n",
       "1      chr17   40064178   40064179  rs61554907\n",
       "2      chr15   90885811   90885812   rs1894401\n",
       "3      chr15   90873319   90873320  rs17514846\n",
       "4      chr15   90880312   90880313      rs6224\n",
       "...      ...        ...        ...         ...\n",
       "20730  chr12    2177796    2177797   rs6489351\n",
       "20737  chr12    2405103    2405104    rs714277\n",
       "20739  chr12    2402664    2402665   rs2239063\n",
       "20745   chr2  161960217  161960218  rs12472555\n",
       "20746   chr2  161989344  161989345   rs2909457\n",
       "\n",
       "[17810 rows x 4 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas_bed = brain_gwas[[\"CHR\", \"BP\", \"SNPS\"]].rename(columns = {\"BP\": \"END\", \"CHR\": \"CHR_ID\"})\n",
    "gwas_bed = gwas_bed[~gwas_bed[\"END\"].isna()] # variants with NaN position likely could not be mapped to hg38 -> discard \n",
    "gwas_bed[\"CHR_ID\"] = gwas_bed[\"CHR_ID\"].apply(lambda x: f\"chr{int(x)}\") \n",
    "gwas_bed[\"START\"] = gwas_bed[\"END\"].apply(lambda x: x - 1) \n",
    "gwas_bed[\"START\"] = gwas_bed[\"START\"].astype(int)\n",
    "gwas_bed[\"END\"] = gwas_bed[\"END\"].astype(int)\n",
    "gwas_bed = gwas_bed[[\"CHR_ID\", \"START\", \"END\", \"SNPS\"]].drop_duplicates()\n",
    "gwas_bed "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "4758a2e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "gwas = BedTool.from_dataframe(gwas_bed).sort()\n",
    "genes = BedTool(genes_bed).sort()\n",
    "closest = gwas.closest(genes, d=True).to_dataframe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "1a4c8df1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t2438058\t2438059\trs4592207\tPLCH2\n",
      " chr1\t2440957\t2440958\trs6688934\tPLCH2\n",
      " chr1\t2441514\t2441515\trs6673661\tPLCH2\n",
      " chr1\t2441728\t2441729\trs6673880\tPLCH2\n",
      " chr1\t2443318\t2443319\trs4648844\tPLCH2\n",
      " chr1\t2444404\t2444405\trs6687012\tPLCH2\n",
      " chr1\t2455661\t2455662\trs4648845\tPLCH2\n",
      " chr1\t8361142\t8361143\trs13596\tRERE\n",
      " chr1\t8362615\t8362616\trs2252865\tRERE\n",
      " chr1\t8363449\t8363450\trs10779702\tRERE\n",
      " "
     ]
    }
   ],
   "source": [
    "closest = closest[[\"chrom\", \"start\", \"end\", \"name\", \"thickEnd\"]]\n",
    "closest = BedTool.from_dataframe(closest)\n",
    "closest.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4a5f3b40",
   "metadata": {},
   "source": [
    "# dmr beds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "c0024fed",
   "metadata": {},
   "outputs": [],
   "source": [
    "bed_files = {f: BedTool(f) for f in glob.glob(f'{input_dir}/*/bed/*.dmr.bed')}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "23150446",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_list = []\n",
    "for bed in bed_files:\n",
    "    output = bed_files[bed].intersect(closest, wb = True)\n",
    "    if len(output) > 1:\n",
    "        tmp = output.to_dataframe()\n",
    "        filename = os.path.basename(bed)\n",
    "        tmp[\"celltype\"] = filename\n",
    "        df_list.append(tmp)\n",
    "\n",
    "dmr_overlap = pd.concat(df_list, ignore_index = True)\n",
    "\n",
    "dmr_overlap = dmr_overlap[[\"chrom\", \"start\", \"end\", \"thickStart\", \"thickEnd\", \"celltype\"]].rename(columns = {\"chrom\": \"CHR_ID\", \n",
    "                                                                                       \"start\" : \"START\", \n",
    "                                                                                       \"end\": \"END\", \n",
    "                                                                                       \"thickStart\": \"SNPS\", \n",
    "                                                                                       \"thickEnd\" : \"Gene\"})\n",
    "dmr_overlap = dmr_overlap.merge(brain_gwas, how = \"left\", on =\"SNPS\")[[\"CHR_ID\", \"START\", \"END\", \"SNPS\", \"Gene\", \n",
    "                                                                       \"trait\", \"celltype\", \"A1\", \"A2\", \"P\"]]\n",
    "\n",
    "dmr_overlap.to_csv('dmr_signif_snps_overlap.tsv', sep = \"\\t\", index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "862bdfab",
   "metadata": {},
   "source": [
    "# cell type specific bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "44a3c6b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# /anvil/projects/x-mcb130189/Wubin/BG/pseudobulk/DMR/Subclass/Neuron/cell_type_specific_bed\n",
    "cell_type_specific_bed_files = {f: BedTool(f) for f in glob.glob(f'{input_dir}/*/cell_type_specific_bed/*.dmr.bed')}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "475213f4",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_list = []\n",
    "for bed in cell_type_specific_bed_files:\n",
    "    output = cell_type_specific_bed_files[bed].intersect(closest, wb = True)\n",
    "    if len(output) > 1:\n",
    "        tmp = output.to_dataframe()\n",
    "        filename = os.path.basename(bed)\n",
    "        tmp[\"celltype\"] = filename\n",
    "        df_list.append(tmp)\n",
    "\n",
    "dmr_overlap = pd.concat(df_list, ignore_index = True)\n",
    "\n",
    "dmr_overlap = dmr_overlap[[\"chrom\", \"start\", \"end\", \"thickStart\", \"thickEnd\", \"celltype\"]].rename(columns = {\"chrom\": \"CHR_ID\", \n",
    "                                                                                       \"start\" : \"START\", \n",
    "                                                                                       \"end\": \"END\", \n",
    "                                                                                       \"thickStart\": \"SNPS\", \n",
    "                                                                                       \"thickEnd\" : \"Gene\"})\n",
    "dmr_overlap = dmr_overlap.merge(brain_gwas, how = \"left\", on =\"SNPS\")[[\"CHR_ID\", \"START\", \"END\", \"SNPS\", \"Gene\", \n",
    "                                                                       \"trait\", \"celltype\", \"A1\", \"A2\", \"P\"]]\n",
    "\n",
    "dmr_overlap.to_csv('ct_specific_dmr_signif_snps_overlap.tsv', sep = \"\\t\", index = False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.20"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
