{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "8f19e8a9",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import pybedtools\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "2ef9efea",
   "metadata": {},
   "outputs": [],
   "source": [
    "# PARAMETERS\n",
    "gwas=\"Table S2. Detailed summary statistics on all nominated risk variants, known and novel_.txt\"\n",
    "bed_dir=\"/anvil/projects/x-mcb130189/Wubin/BG/pseudobulk/DMR/Subclass/Neuron/bed\"\n",
    "plink_dir=\"/anvil/projects/x-mcb130189/rwang22/bican/ref/GRCh38/plink_files\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2e76d6d8",
   "metadata": {},
   "source": [
    "# filter to get the 90 significant loci \n",
    "These additional filters exclude variants that 1) had a random-effects P value across all datasets > 4.67E-04 and 2) a conditional analysis P > 4.67E-04 using participant level 23andMe genotype data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b1eba992",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP</th>\n",
       "      <th>Nearest Gene</th>\n",
       "      <th>QTL Nominated Gene (nearest QTL)</th>\n",
       "      <th>Effect allele</th>\n",
       "      <th>Other allele</th>\n",
       "      <th>Effect allele frequency</th>\n",
       "      <th>Beta, all studies</th>\n",
       "      <th>SE, all studies</th>\n",
       "      <th>...</th>\n",
       "      <th>Freq1, new studies</th>\n",
       "      <th>Beta, new studies</th>\n",
       "      <th>StdErr, new studies</th>\n",
       "      <th>P, new studies</th>\n",
       "      <th>I2, new studies</th>\n",
       "      <th>Passes pooled 23andMe QC</th>\n",
       "      <th>Known GWAS locus within 1MB</th>\n",
       "      <th>Failed final filtering and QC</th>\n",
       "      <th>Locus within 250KB</th>\n",
       "      <th>Locus Number</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs114138760</td>\n",
       "      <td>1</td>\n",
       "      <td>154,898,185</td>\n",
       "      <td>PMVK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c</td>\n",
       "      <td>g</td>\n",
       "      <td>0.0112</td>\n",
       "      <td>0.2812</td>\n",
       "      <td>0.0478</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0109</td>\n",
       "      <td>0.1997</td>\n",
       "      <td>0.0843</td>\n",
       "      <td>1.780000e-02</td>\n",
       "      <td>0.0</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs35749011</td>\n",
       "      <td>1</td>\n",
       "      <td>155,135,036</td>\n",
       "      <td>KRTCAP2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.0169</td>\n",
       "      <td>0.6068</td>\n",
       "      <td>0.0342</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0177</td>\n",
       "      <td>0.6798</td>\n",
       "      <td>0.0615</td>\n",
       "      <td>2.160000e-28</td>\n",
       "      <td>0.0</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs76763715</td>\n",
       "      <td>1</td>\n",
       "      <td>155,205,634</td>\n",
       "      <td>GBAP1</td>\n",
       "      <td>GBAP1</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.9953</td>\n",
       "      <td>-0.7467</td>\n",
       "      <td>0.0765</td>\n",
       "      <td>...</td>\n",
       "      <td>0.9941</td>\n",
       "      <td>-0.6693</td>\n",
       "      <td>0.1269</td>\n",
       "      <td>1.340000e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>F</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs6658353</td>\n",
       "      <td>1</td>\n",
       "      <td>161,469,054</td>\n",
       "      <td>FCGR2A</td>\n",
       "      <td>FCGR2A</td>\n",
       "      <td>c</td>\n",
       "      <td>g</td>\n",
       "      <td>0.5011</td>\n",
       "      <td>0.0650</td>\n",
       "      <td>0.0094</td>\n",
       "      <td>...</td>\n",
       "      <td>0.5105</td>\n",
       "      <td>0.0662</td>\n",
       "      <td>0.0171</td>\n",
       "      <td>1.100000e-04</td>\n",
       "      <td>46.4</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs11578699</td>\n",
       "      <td>1</td>\n",
       "      <td>171,719,769</td>\n",
       "      <td>VAMP4</td>\n",
       "      <td>VAMP4</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.1949</td>\n",
       "      <td>-0.0704</td>\n",
       "      <td>0.0120</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1947</td>\n",
       "      <td>-0.0663</td>\n",
       "      <td>0.0226</td>\n",
       "      <td>3.310000e-03</td>\n",
       "      <td>13.4</td>\n",
       "      <td>T</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           SNP  CHR           BP Nearest Gene  \\\n",
       "0  rs114138760    1  154,898,185         PMVK   \n",
       "1   rs35749011    1  155,135,036      KRTCAP2   \n",
       "2   rs76763715    1  155,205,634        GBAP1   \n",
       "3    rs6658353    1  161,469,054       FCGR2A   \n",
       "4   rs11578699    1  171,719,769        VAMP4   \n",
       "\n",
       "  QTL Nominated Gene (nearest QTL) Effect allele Other allele  \\\n",
       "0                              NaN             c            g   \n",
       "1                              NaN             a            g   \n",
       "2                            GBAP1             t            c   \n",
       "3                           FCGR2A             c            g   \n",
       "4                            VAMP4             t            c   \n",
       "\n",
       "   Effect allele frequency  Beta, all studies  SE, all studies  ...  \\\n",
       "0                   0.0112             0.2812           0.0478  ...   \n",
       "1                   0.0169             0.6068           0.0342  ...   \n",
       "2                   0.9953            -0.7467           0.0765  ...   \n",
       "3                   0.5011             0.0650           0.0094  ...   \n",
       "4                   0.1949            -0.0704           0.0120  ...   \n",
       "\n",
       "   Freq1, new studies  Beta, new studies  StdErr, new studies  P, new studies  \\\n",
       "0              0.0109             0.1997               0.0843    1.780000e-02   \n",
       "1              0.0177             0.6798               0.0615    2.160000e-28   \n",
       "2              0.9941            -0.6693               0.1269    1.340000e-07   \n",
       "3              0.5105             0.0662               0.0171    1.100000e-04   \n",
       "4              0.1947            -0.0663               0.0226    3.310000e-03   \n",
       "\n",
       "   I2, new studies  Passes pooled 23andMe QC  Known GWAS locus within 1MB  \\\n",
       "0              0.0                         T                            1   \n",
       "1              0.0                         T                            1   \n",
       "2              0.0                         F                            1   \n",
       "3             46.4                         T                            0   \n",
       "4             13.4                         T                            0   \n",
       "\n",
       "   Failed final filtering and QC  Locus within 250KB  Locus Number  \n",
       "0                              0                 1.0           1.0  \n",
       "1                              0                 1.0           1.0  \n",
       "2                              0                 1.0           1.0  \n",
       "3                              0                 0.0           2.0  \n",
       "4                              0                 0.0           3.0  \n",
       "\n",
       "[5 rows x 31 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_variants = pd.read_csv(gwas, sep = \"\\t\")\n",
    "all_variants.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "67373e1b",
   "metadata": {},
   "outputs": [],
   "source": [
    "top_90 = all_variants[~all_variants[\"Locus Number\"].isna()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6e59de4a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CAN NOT FIND rs76763715\n",
      "CAN NOT FIND rs34637584\n"
     ]
    }
   ],
   "source": [
    "# liftover to hg38\n",
    "rows = []\n",
    "for index, row in top_90.iterrows():\n",
    "    chr = row[\"CHR\"]\n",
    "    bim_file = f'{plink_dir}/1000G.EUR.hg38.{chr}.bim'\n",
    "    bim = pd.read_csv(bim_file, sep = \"\\t\", header = None, \n",
    "                      names=[\"CHR\", \"SNP\", \"GD\", \"BP\", \"A1\", \"A2\"])\n",
    "    cur = bim[bim[\"SNP\"] == row[\"SNP\"]]\n",
    "    if len(cur) == 0:\n",
    "        print(f'CAN NOT FIND {row[\"SNP\"]}')\n",
    "    else:\n",
    "        bp = int(row[\"BP\"].replace(',', ''))\n",
    "        rows.append({})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "804b5c76",
   "metadata": {},
   "outputs": [],
   "source": [
    "rows.append({'SNP' : \"rs76763715\", 'CHR': 1, 'BP': 155235843})\n",
    "rows.append({'SNP' : \"rs34637584\", 'CHR': 12, 'BP': 40340400})\n",
    "df = pd.DataFrame(rows)\n",
    "df[\"START\"] = df[\"BP\"] - 1\n",
    "df = df[['CHR', 'START', 'BP', 'SNP']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "5e4f1d74",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>START</th>\n",
       "      <th>BP</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>154898184</td>\n",
       "      <td>154898185</td>\n",
       "      <td>rs114138760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>155135035</td>\n",
       "      <td>155135036</td>\n",
       "      <td>rs35749011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>161469053</td>\n",
       "      <td>161469054</td>\n",
       "      <td>rs6658353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>171719768</td>\n",
       "      <td>171719769</td>\n",
       "      <td>rs11578699</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>205723571</td>\n",
       "      <td>205723572</td>\n",
       "      <td>rs823118</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR      START         BP          SNP\n",
       "0    1  154898184  154898185  rs114138760\n",
       "1    1  155135035  155135036   rs35749011\n",
       "2    1  161469053  161469054    rs6658353\n",
       "3    1  171719768  171719769   rs11578699\n",
       "4    1  205723571  205723572     rs823118"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "93c97b86",
   "metadata": {},
   "source": [
    "# bed file intersect with hypo dmrs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "e8c70ed7",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "***** WARNING: File /tmp/pybedtools.5q8osvh6.tmp has inconsistent naming convention for record:\n",
      "chr1\t53062\t53562\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.5q8osvh6.tmp has inconsistent naming convention for record:\n",
      "chr1\t53062\t53562\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.py14hsb4.tmp has inconsistent naming convention for record:\n",
      "chr1\t780748\t781248\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.py14hsb4.tmp has inconsistent naming convention for record:\n",
      "chr1\t780748\t781248\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.uaghexio.tmp has inconsistent naming convention for record:\n",
      "chr1\t19142\t19642\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.uaghexio.tmp has inconsistent naming convention for record:\n",
      "chr1\t19142\t19642\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.zq8b4bz7.tmp has inconsistent naming convention for record:\n",
      "chr1\t19142\t19642\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.zq8b4bz7.tmp has inconsistent naming convention for record:\n",
      "chr1\t19142\t19642\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.kjhnni4m.tmp has inconsistent naming convention for record:\n",
      "chr1\t31721\t32221\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.kjhnni4m.tmp has inconsistent naming convention for record:\n",
      "chr1\t31721\t32221\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.3h67n_t7.tmp has inconsistent naming convention for record:\n",
      "chr1\t15023\t15523\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.3h67n_t7.tmp has inconsistent naming convention for record:\n",
      "chr1\t15023\t15523\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.evbh1ny2.tmp has inconsistent naming convention for record:\n",
      "chr1\t40166\t40666\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.evbh1ny2.tmp has inconsistent naming convention for record:\n",
      "chr1\t40166\t40666\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.4cxefon2.tmp has inconsistent naming convention for record:\n",
      "chr1\t13998\t14498\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.4cxefon2.tmp has inconsistent naming convention for record:\n",
      "chr1\t13998\t14498\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.cb75u8cj.tmp has inconsistent naming convention for record:\n",
      "chr1\t53718\t54218\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.cb75u8cj.tmp has inconsistent naming convention for record:\n",
      "chr1\t53718\t54218\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools._b3pwvxb.tmp has inconsistent naming convention for record:\n",
      "chr1\t89782\t90282\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools._b3pwvxb.tmp has inconsistent naming convention for record:\n",
      "chr1\t89782\t90282\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.8lriws5k.tmp has inconsistent naming convention for record:\n",
      "chr1\t13227\t13727\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.8lriws5k.tmp has inconsistent naming convention for record:\n",
      "chr1\t13227\t13727\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.3oapckbi.tmp has inconsistent naming convention for record:\n",
      "chr1\t53062\t53562\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.3oapckbi.tmp has inconsistent naming convention for record:\n",
      "chr1\t53062\t53562\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.sz3qv_mx.tmp has inconsistent naming convention for record:\n",
      "chr1\t15023\t15523\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.sz3qv_mx.tmp has inconsistent naming convention for record:\n",
      "chr1\t15023\t15523\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.6lx3i9f9.tmp has inconsistent naming convention for record:\n",
      "chr1\t13227\t13727\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.6lx3i9f9.tmp has inconsistent naming convention for record:\n",
      "chr1\t13227\t13727\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.7tb15851.tmp has inconsistent naming convention for record:\n",
      "chr1\t13227\t13727\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.7tb15851.tmp has inconsistent naming convention for record:\n",
      "chr1\t13227\t13727\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.pkwe5dg0.tmp has inconsistent naming convention for record:\n",
      "chr1\t13998\t14498\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.pkwe5dg0.tmp has inconsistent naming convention for record:\n",
      "chr1\t13998\t14498\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.jyxgxlr4.tmp has inconsistent naming convention for record:\n",
      "chr1\t13998\t14498\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.jyxgxlr4.tmp has inconsistent naming convention for record:\n",
      "chr1\t13998\t14498\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.34xqd5i7.tmp has inconsistent naming convention for record:\n",
      "chr1\t53718\t54218\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.34xqd5i7.tmp has inconsistent naming convention for record:\n",
      "chr1\t53718\t54218\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.wozi4xm8.tmp has inconsistent naming convention for record:\n",
      "chr1\t13227\t13727\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.wozi4xm8.tmp has inconsistent naming convention for record:\n",
      "chr1\t13227\t13727\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.1z24evzn.tmp has inconsistent naming convention for record:\n",
      "chr1\t25787\t26287\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.1z24evzn.tmp has inconsistent naming convention for record:\n",
      "chr1\t25787\t26287\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.atojh4ws.tmp has inconsistent naming convention for record:\n",
      "chr1\t19142\t19642\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.atojh4ws.tmp has inconsistent naming convention for record:\n",
      "chr1\t19142\t19642\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.uumcl8vf.tmp has inconsistent naming convention for record:\n",
      "chr1\t32675\t33175\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.uumcl8vf.tmp has inconsistent naming convention for record:\n",
      "chr1\t32675\t33175\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.g1b88exv.tmp has inconsistent naming convention for record:\n",
      "chr1\t20753\t21253\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.g1b88exv.tmp has inconsistent naming convention for record:\n",
      "chr1\t20753\t21253\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.bw8xa2k4.tmp has inconsistent naming convention for record:\n",
      "chr1\t19142\t19642\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.bw8xa2k4.tmp has inconsistent naming convention for record:\n",
      "chr1\t19142\t19642\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.7zegdojb.tmp has inconsistent naming convention for record:\n",
      "chr1\t19934\t20434\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.7zegdojb.tmp has inconsistent naming convention for record:\n",
      "chr1\t19934\t20434\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.6nk_mcwi.tmp has inconsistent naming convention for record:\n",
      "chr1\t19289\t19789\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.6nk_mcwi.tmp has inconsistent naming convention for record:\n",
      "chr1\t19289\t19789\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.46rjfwkv.tmp has inconsistent naming convention for record:\n",
      "chr1\t53718\t54218\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.46rjfwkv.tmp has inconsistent naming convention for record:\n",
      "chr1\t53718\t54218\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.b589n9mz.tmp has inconsistent naming convention for record:\n",
      "chr1\t13227\t13727\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.b589n9mz.tmp has inconsistent naming convention for record:\n",
      "chr1\t13227\t13727\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools._8im0xom.tmp has inconsistent naming convention for record:\n",
      "chr1\t32675\t33175\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools._8im0xom.tmp has inconsistent naming convention for record:\n",
      "chr1\t32675\t33175\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.ky39cdt2.tmp has inconsistent naming convention for record:\n",
      "chr1\t24661\t25161\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.ky39cdt2.tmp has inconsistent naming convention for record:\n",
      "chr1\t24661\t25161\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.vqi5x7w1.tmp has inconsistent naming convention for record:\n",
      "chr1\t19289\t19789\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.vqi5x7w1.tmp has inconsistent naming convention for record:\n",
      "chr1\t19289\t19789\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.rvtx4l10.tmp has inconsistent naming convention for record:\n",
      "chr1\t53718\t54218\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.rvtx4l10.tmp has inconsistent naming convention for record:\n",
      "chr1\t53718\t54218\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.nx62qqse.tmp has inconsistent naming convention for record:\n",
      "chr1\t261404\t261904\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.nx62qqse.tmp has inconsistent naming convention for record:\n",
      "chr1\t261404\t261904\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.89bdim3l.tmp has inconsistent naming convention for record:\n",
      "chr1\t13227\t13727\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.89bdim3l.tmp has inconsistent naming convention for record:\n",
      "chr1\t13227\t13727\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.deiqg5b2.tmp has inconsistent naming convention for record:\n",
      "chr1\t15023\t15523\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.deiqg5b2.tmp has inconsistent naming convention for record:\n",
      "chr1\t15023\t15523\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.7mweant6.tmp has inconsistent naming convention for record:\n",
      "chr1\t24661\t25161\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.7mweant6.tmp has inconsistent naming convention for record:\n",
      "chr1\t24661\t25161\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.9i4500_h.tmp has inconsistent naming convention for record:\n",
      "chr1\t261404\t261904\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.9i4500_h.tmp has inconsistent naming convention for record:\n",
      "chr1\t261404\t261904\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.pgwm0510.tmp has inconsistent naming convention for record:\n",
      "chr1\t261404\t261904\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.pgwm0510.tmp has inconsistent naming convention for record:\n",
      "chr1\t261404\t261904\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.abdea2sz.tmp has inconsistent naming convention for record:\n",
      "chr1\t19289\t19789\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.abdea2sz.tmp has inconsistent naming convention for record:\n",
      "chr1\t19289\t19789\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.btlkqrq0.tmp has inconsistent naming convention for record:\n",
      "chr1\t78451\t78951\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.btlkqrq0.tmp has inconsistent naming convention for record:\n",
      "chr1\t78451\t78951\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.k7drt1zy.tmp has inconsistent naming convention for record:\n",
      "chr1\t25724\t26224\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.k7drt1zy.tmp has inconsistent naming convention for record:\n",
      "chr1\t25724\t26224\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.td04hi5q.tmp has inconsistent naming convention for record:\n",
      "chr1\t53718\t54218\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.td04hi5q.tmp has inconsistent naming convention for record:\n",
      "chr1\t53718\t54218\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.eefc4dv8.tmp has inconsistent naming convention for record:\n",
      "chr1\t29905\t30405\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.eefc4dv8.tmp has inconsistent naming convention for record:\n",
      "chr1\t29905\t30405\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.k18_dt_s.tmp has inconsistent naming convention for record:\n",
      "chr1\t13227\t13727\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.k18_dt_s.tmp has inconsistent naming convention for record:\n",
      "chr1\t13227\t13727\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.v79woz19.tmp has inconsistent naming convention for record:\n",
      "chr1\t257621\t258121\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.v79woz19.tmp has inconsistent naming convention for record:\n",
      "chr1\t257621\t258121\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.910p7udt.tmp has inconsistent naming convention for record:\n",
      "chr1\t53062\t53562\n",
      "\n",
      "***** WARNING: File /tmp/pybedtools.910p7udt.tmp has inconsistent naming convention for record:\n",
      "chr1\t53062\t53562\n",
      "\n"
     ]
    }
   ],
   "source": [
    "hits = []\n",
    "for bedfile in os.listdir(bed_dir):\n",
    "    x = pd.read_csv(f'{bed_dir}/{bedfile}', header = None, sep = \"\\t\", names = [\"chr\", \"start\", \"end\"])\n",
    "    dmr = pybedtools.BedTool.from_dataframe(x)\n",
    "    # print(dmr.head())\n",
    "    snp = pybedtools.BedTool.from_dataframe(df)\n",
    "    tmp = dmr.intersect(snp)\n",
    "    if tmp.count() > 0:\n",
    "        hits.append(bedfile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "98842355",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hits"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.20"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
